### PR TITLE
Compiler 💅

### PR DESCRIPTION
**Pull request by rtimmons**
Gets rid of some compiler and clang-tidy errors.

### CODE REVIEW

File: run-genny [hunk 1]
```diff
@@ -124,6 +125,7 @@ if [[ ! -d "${GENNY_REPO_ROOT}/${_genny_venv}" || ! -e "${GENNY_REPO_ROOT}/${_ge
     source "${GENNY_REPO_ROOT}/${_genny_venv}/bin/activate"
 
     # Install Genny requirements
+    _pip install "numpy==1.22.3"  # For some reason pip doesn't pick this up in requirements.txt?
```
Comment (jimoleary) [inline run-genny:128]: you sure? This worked in a clean /old env for me:

```
$ genny_venv/bin/pip show numpy
Name: numpy
Version: 1.22.3
Summary: NumPy is the fundamental package for array computing with Python.
Home-page: https://www.numpy.org
Author: Travis E. Oliphant et al.
Author-email: None
License: BSD
Location: /home/jim/git/genny-trees/master/genny_venv/lib/python3.9/site-packages
Requires: 
Required-by: numexpr

```

File: src/cast_core/src/CollectionScanner.cpp [hunk 2]
```diff
@@ -222,7 +222,7 @@ auto evaluatePipeline(CollectionScanner::PhaseConfig* config) {
     auto expr = config->aggregatePipelineExpr->evaluate();
     bsoncxx::array::view pipeline{expr.view()["array"].get_array().value};
     mongocxx::pipeline pl{};
-    pl.append_stages(std::move(pipeline));
+    pl.append_stages(pipeline);
```
Comment (jimoleary) [inline src/cast_core/src/CollectionScanner.cpp:225]: Is this from a warning? The  mongocxx driver tests have a [std::move](https://github.com/mongodb/mongo-cxx-driver/blob/7006baba22cfdcd43a45b7298d3f66ec5f67c25b/src/mongocxx/test/spec/operation.cpp#L64-L70)?

File: src/cast_core/src/CollectionScanner.cpp [hunk 3]
```diff
@@ -508,11 +507,13 @@ void CollectionScanner::run() {
 
 CollectionScanner::CollectionScanner(genny::ActorContext& context)
     : Actor{context},
+      // TODO: _totalInserts is unused
```
Comment (jimoleary) [inline src/cast_core/src/CollectionScanner.cpp:510]: Do you want a ticket number for the TODO (same question for line 516)

File: src/gennylib/test/context_test.cpp [hunk 4]
```diff
@@ -43,7 +38,11 @@ using Catch::Matchers::StartsWith;
 // The driver checks the passed-in mongo uri for accuracy but doesn't actually
 // initiate a connection until a connection is retrieved from
 // the connection-pool
-static const std::string baseYaml = "SchemaVersion: 2018-07-01\nClients: {Default: {URI: 'mongodb://localhost:27017'}}\nActors: []\n";
+static const std::string baseYaml() {
```
Comment (jimoleary) [inline src/gennylib/test/context_test.cpp:41]: why bother change this ?

File: src/lamplib/requirements.txt [hunk 5]
```diff
@@ -12,4 +13,5 @@ shrub.py==1.1.0
 black==19.10b0
 setuptools==51.0.0
 pytest==6.0.2
+numpy==1.22.3  # again
```
Comment (jimoleary) [inline src/lamplib/requirements.txt:16]: this can't be right?

File: src/lamplib/src/genny/tasks/dry_run.py [hunk 6]
```diff
@@ -32,6 +32,9 @@ def dry_run_workload(
         "AuthNInsert.yml",
         "ParallelInsert.yml",
         "LoggingActorExample.yml",
+        "Mixed10KThreads.yml",
```
Comment (jimoleary) [inline src/lamplib/src/genny/tasks/dry_run.py:35]: This doesn't seem like a scaleable solution, would it make more sense to get this info directly from the workload file?

File: src/third_party/cppcodec/CMakeLists.txt [hunk 7]
```diff
@@ -1,6 +1,6 @@
-cmake_minimum_required(VERSION 2.8.5)
-project(cppcodec CXX)
-set(PROJECT_VERSION 0.1)
+cmake_minimum_required(VERSION 2.8.12)
+cmake_policy(SET CMP0048 NEW)
```
Comment (jimoleary) [inline src/third_party/cppcodec/CMakeLists.txt:2]: It seems that [CMP0048](https://cmake.org/cmake/help/latest/policy/CMP0048.html) was introduced in cmake 3.0. Does that mean that the cmake_minimum_required should be higher?

File: src/third_party/cppcodec/CMakeLists.txt [hunk 8]
```diff
@@ -9,26 +9,30 @@ include(CTest)
 # In your own project that uses cppcodec, you might want to specify a different standard or error level.
 
 # Request C++11, or let the user specify the standard on via -D command line option.
-if (NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 11)
-endif()
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+#if (NOT CMAKE_CXX_STANDARD)
```
Comment (jimoleary) [inline src/third_party/cppcodec/CMakeLists.txt:12]: should most of this be removed if it is not needed?

File: src/third_party/cppcodec/CMakeLists.txt [hunk 9]
```diff
@@ -63,12 +67,12 @@ set(PUBLIC_HEADERS
 
 add_library(cppcodec OBJECT ${PUBLIC_HEADERS}) # unnecessary for building, but makes headers show up in IDEs
 set_target_properties(cppcodec PROPERTIES LINKER_LANGUAGE CXX)
-add_subdirectory(tool)
-add_subdirectory(example)
+# add_subdirectory(tool)
```
Comment (jimoleary) [inline src/third_party/cppcodec/CMakeLists.txt:70]: remove?

File: src/third_party/simple-beast-client/CMakeLists.txt [hunk 10]
```diff
@@ -7,7 +7,8 @@ find_package(Boost 1.66 REQUIRED COMPONENTS
 find_package(OpenSSL)
 find_package(Threads)
 find_package(cppcodec)
-target_include_directories(simple-beast-client INTERFACE
+# SYSTEM INTERFACE means we don't emit warnings.
```
Comment (jimoleary) [inline src/third_party/simple-beast-client/CMakeLists.txt:10]: nit: it's just system that [supresses warnings](https://cmake.org/cmake/help/latest/command/target_include_directories.html).

And is that a good idea?

File: src/value_generators/src/DocumentGenerator.cpp [hunk 11]
```diff
@@ -65,12 +63,16 @@ class Generator : public Appendable {
 
 namespace {
 using namespace genny;
-const static boost::posix_time::ptime epoch{boost::gregorian::date(1970, 1, 1)};
+
```
Comment (jimoleary) [inline src/value_generators/src/DocumentGenerator.cpp:66]: I feel that I have somewhat contributed to some of the formatting issues here. Will evergreen.yml give errors for these issues in future?

**Review by jimoleary — COMMENTED** [review]
Some minorish questions.

File: run-genny (hunk 1 above)
Comment (rtimmons) [reply run-genny:128]: I fought this for over an hour and eventually kinda threw my hands up. What worked on macOS didn't work on Ubuntu and vice-versa, and installing on macOS makes it compile numpy from source in many circumstances. I lost my will to make this better.

File: src/cast_core/src/CollectionScanner.cpp (hunk 2 above)
Comment (rtimmons) [reply src/cast_core/src/CollectionScanner.cpp:225]: Ya it's a clang-tidy warning:

> Clang-Tidy: Std::move of the variable 'pipeline' of the trivially-copyable type 'bsoncxx::array::view' has no effect; remove std::move()

I don't know enough about clang tooling to turn these on in CI/lint, but CLion helpfully points them out. I tried to remove as many as possible.

File: src/cast_core/src/CollectionScanner.cpp (hunk 3 above)
Comment (rtimmons) [reply src/cast_core/src/CollectionScanner.cpp:510]: In reality I don't suspect that a ticket would ever get prioritized so would just be noise. This Actor is owned by Storage Engines. I think this comment is only useful if/when this Actor gets any attention from future maintainers; this change isn't all that important on its own.

File: src/gennylib/test/context_test.cpp (hunk 4 above)
Comment (rtimmons) [reply src/gennylib/test/context_test.cpp:41]: It's another clang-tidy thing:

> Clang-Tidy: Initialization of 'baseYaml' with static storage duration may throw an exception that cannot be caught

File: src/lamplib/requirements.txt (hunk 5 above)
Comment (rtimmons) [reply src/lamplib/requirements.txt:16]: Another place where I lost my will to fight it. Ideally we'd just move the python to poetry.

File: src/third_party/cppcodec/CMakeLists.txt (hunk 8 above)
Comment (rtimmons) [reply src/third_party/cppcodec/CMakeLists.txt:12]: This file is mostly copy/pasted (vendored) from cppcodec, and I figured leaving the lines commented-out would help any poor souls trying to update cppcodec in the future identify what things were changed.

File: src/third_party/simple-beast-client/CMakeLists.txt (hunk 10 above)
Comment (rtimmons) [reply src/third_party/simple-beast-client/CMakeLists.txt:10]: Given inifinite time/resources we'd have this as a part of the genny toolchain (precompiled) and/or we'd fix the issue in the 3p lib and upstream it. As it is, there is nothing genny maintainers are expected to do with compiler warnigns coming from simple-beast-client.

**Comment by rtimmons** [comment]
Thanks for your as usual thorough and helpful review, @jimoleary. I made some changes but mostly punted on anything substantial. Please push back wherever you disagree.

**Review by jimoleary — APPROVED** [review]
lgtm.

If you know what you did to make clang-tidy work on clion that would be great to share otherwise all good.

Thanks!

File: src/cast_core/src/CollectionScanner.cpp (hunk 2 above)
Comment (rtimmons) [reply src/cast_core/src/CollectionScanner.cpp:225]: I don't have any `.clang-tidy` configs afaict. Do you have the Inspection enabled?

Screen Shot 2022-05-16 at 1 18 59 PM

**Comment by rtimmons** [comment]
evergreen merge

**Pull request merged**